src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -66,39 +78,81 @@ const TopicFragment = ({
+  const exhangeFollower = (newTopicLists, willFollow, topicId) => {
+    const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+    newTopicLists[indexTopic].following = !!willFollow;
+    newTopicLists[indexTopic].user_id_follower = myId;
+    return newTopicLists[indexTopic];
+  };
+
+  const handleTopic = (from, willFollow, item, index) => {
     if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
-      DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
+
+      DiscoveryAction.setDiscoveryInitialTopics(
+        discovery.initialTopics.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        }),
+        discoveryDispatch
+      );
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
 
-      DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
+      DiscoveryAction.setDiscoveryInitialTopics(
+        discovery.initialTopics.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        }),
+        discoveryDispatch
+      );
     }
 
     if (from === FROM_FOLLOWED_TOPIC) {
-      const newFollowedTopics = [...followedTopic];
-      newFollowedTopics[index].following = !!willFollow;
-      newFollowedTopics[index].user_id_follower = myId;
+      const newFollowedTopics = [...newMapFollowedTopics];
+      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
 
-      setFollowedTopic(newFollowedTopics);
+      setFollowedTopic(
+        followedTopic.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        })
+      );
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC) {
-      const newUnFollowedTopic = [...unfollowedTopic];
-      newUnFollowedTopic[index].following = !!willFollow;
-      newUnFollowedTopic[index].user_id_follower = myId;
+      const newUnFollowedTopic = [...newMapUnfollowedTopics];
+      const newTopic = exhangeFollower(newUnFollowedTopic, willFollow, item.topic_id);
 
-      setUnfollowedTopic(newUnFollowedTopic);
+      setUnfollowedTopic(
+        unfollowedTopic.map((topic) => {
+          if (topic.topic_id === newTopic.topic_id) {
+            return newTopic;
+          }
+          return topic;
+        })
+      );
     }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleTopic` function has a lot of repeated code. The same logic is applied for different conditions (`FROM_FOLLOWED_TOPIC_INITIAL`, `FROM_UNFOLLOWED_TOPIC_INITIAL`, `FROM_FOLLOWED_TOPIC`, `FROM_UNFOLLOWED_TOPIC`). This violates the DRY (Don't Repeat Yourself) principle. Consider refactoring this function to reduce redundancy and improve maintainability.
```diff
-  const handleTopic = (from, willFollow, item, index) => {
-    if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
-      const newFollowedTopics = [...topics];
-      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
-      DiscoveryAction.setDiscoveryInitialTopics(
-        discovery.initialTopics.map((topic) => {
-          if (topic.topic_id === newTopic.topic_id) {
-            return newTopic;
-          }
-          return topic;
-        }),
-        discoveryDispatch
-      );
-    }
-    // similar code for other conditions...
-  };
+  const handleTopic = (from, willFollow, item, index) => {
+    let newTopics;
+    let setFunction;
+    switch(from) {
+      case FROM_FOLLOWED_TOPIC_INITIAL:
+      case FROM_UNFOLLOWED_TOPIC_INITIAL:
+        newTopics = [...topics];
+        setFunction = DiscoveryAction.setDiscoveryInitialTopics;
+        break;
+      case FROM_FOLLOWED_TOPIC:
+        newTopics = [...newMapFollowedTopics];
+        setFunction = setFollowedTopic;
+        break;
+      case FROM_UNFOLLOWED_TOPIC:
+        newTopics = [...newMapUnfollowedTopics];
+        setFunction = setUnfollowedTopic;
+        break;
+    }
+    const newTopic = exhangeFollower(newTopics, willFollow, item.topic_id);
+    setFunction(
+      newTopics.map((topic) => {
+        if (topic.topic_id === newTopic.topic_id) {
+          return newTopic;
+        }
+        return topic;
+      })
+    );
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->